libdnf5/system/state.cpp

```diff
@@ -607,9 +645,48 @@ static T load_toml_data(const std::string & path, const std::string & key) {
 }
 
 
+// In a given directory find all files endig with ".new" suffix and return them
+static std::vector<std::filesystem::path> gather_suffix_new_files(const std::filesystem::path & directory) {
```

Comment:
One more thing – all `std::filesystem::exists()`, `std::filesystem::directory_iterator()`, `std::filesystem::is_directory()`, and especially `is_regular_file()` can throw an exception (e.g., if the directory does not exist or permission is denied), which will result in a `StateLoadError` being thrown. In most cases, this is fine and expected, but, for example, the presence of any inaccessible file (not even one of the `*.new` or `*.toml` files) in the system state directory would unnecessarily prevent the system state from loading.

Reply:
> the presence of any inaccessible file (not even one of the *.new or *.toml files)

I might not understand it enough but how could that happen?

If a user has the privileges to read the system state directory (which is required) then they see all the contents (basically have access to all the information in https://en.cppreference.com/w/cpp/filesystem/directory_entry.html) no?

Although one similar case is when we have some `*.new` files (the previous system state update failed) and an unprivileged user tries to read it. It would fail on the remove or rename, I have added a log for that.